pkg/webhooks/scc/scc.go

```diff
@@ -0,0 +1,208 @@
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{"security.openshift.io"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"securitycontextconstraints"},
+				Scope:       &scope,
+			},
+		},
+	}
+	defaultSCCs = []string{
+		"anyuid",
+		"hostaccess",
+		"hostmount-anyuid",
+		"hostnetwork",
+		"node-exporter",
+		"nonroot",
+		"privileged",
+		"restricted",
+		"pipelines-scc",
+	}
+)
+
+type SCCWebHook struct {
+	s runtime.Scheme
+}
+
+// NewWebhook creates the new webhook
+func NewWebhook() *SCCWebHook {
+	scheme := runtime.NewScheme()
+	admissionv1.AddToScheme(scheme)
+	corev1.AddToScheme(scheme)
+
+	return &SCCWebHook{
+		s: *scheme,
+	}
+}
+
+// Authorized implements Webhook interface
+func (s *SCCWebHook) Authorized(request admissionctl.Request) admissionctl.Response {
+	return s.authorized(request)
+}
+
+func (s *SCCWebHook) authorized(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+
+	scc, err := s.renderSCC(request)
+	if err != nil {
+		log.Error(err, "Couldn't render a SCC from the incoming request")
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+
+	if request.Operation == admissionv1.Delete {
+		if isDefaultSCC(scc) {
+			ret = admissionctl.Denied("Deleting default SCCs is not allowed")
+			ret.UID = request.AdmissionRequest.UID
+			return ret
+		}
+	}
+
+	if request.Operation == admissionv1.Update {
+		if isDefaultSCC(scc) {
+			ret = admissionctl.Denied("Modifying default SCCs is not allowed")
+			ret.UID = request.AdmissionRequest.UID
+			return ret
+		}
+	}
+
+	if request.Operation == admissionv1.Create {
+		if SCCwithHigherPriority(scc) {
+			ret = admissionctl.Denied(fmt.Sprintf("Creating SCC with priority higher than %d is not allowed", anyuidPriority))
+			ret.UID = request.AdmissionRequest.UID
+			return ret
+		}
+	}
+
+	ret = admissionctl.Allowed("Request is allowed")
+	ret.UID = request.AdmissionRequest.UID
+	return ret
+}
+
+// renderSCC render the SCC object from the requests
+func (s *SCCWebHook) renderSCC(request admissionctl.Request) (*securityv1.SecurityContextConstraints, error) {
+	decoder, err := admissionctl.NewDecoder(&s.s)
+	if err != nil {
+		return nil, err
+	}
+	scc := &securityv1.SecurityContextConstraints{}
+	if len(request.OldObject.Raw) > 0 {
+		err = decoder.DecodeRaw(request.OldObject, scc)
+	} else {
+		err = decoder.DecodeRaw(request.Object, scc)
+	}
+	if err != nil {
+		return nil, err
+	}
+	return scc, nil
+}
+
+// isDefaultSCC checks if the request is going to operate on the SCC in the
+// default list
+func isDefaultSCC(scc *securityv1.SecurityContextConstraints) bool {
+	for _, s := range defaultSCCs {
+		if scc.Name == s {
+			return true
+		}
+	}
+	return false
+}
+
+// SCCwithHigherPriority checks if the created SCC has the higher priority
+// than 10 (default to anyuid)
+func SCCwithHigherPriority(scc *securityv1.SecurityContextConstraints) bool {
+	if scc.Priority != nil {
+		if *scc.Priority > anyuidPriority {
```

Comment:
can you elaborate why it's needed to prevent customers from creating SCCs with higher priority than anyuid? I think it's not described in the card. I assume it is so the default used if none is specified doesn't change, but I'm not 100% sure?

Reply:
We were seeing problem due to the customer created higher priority SCC and the `anyuid` cannot be applied to pods . see OHSS-7388
I am assuming this might be a potential issue which we can restrict via webhook. WDYT?